packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -606,19 +608,19 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with AutomaticK
         // First add the explicit suffix if the suffix visibility mode matches.
         if (_showSuffixWidget(text)) {
           rowChildren.add(widget.suffix);
-        // Otherwise, try to show a clear button if its visibility mode matches.
+          // Otherwise, try to show a clear button if its visibility mode matches.
         } else if (_showClearButton(text)) {
           rowChildren.add(
             GestureDetector(
               onTap: widget.enabled ?? true
                   ? () => _effectiveController.clear()
                   : null,
-              child: const Padding(
-                padding: EdgeInsets.symmetric(horizontal: 6.0),
+              child: Padding(
+                padding: const EdgeInsets.symmetric(horizontal: 6.0),
                 child: Icon(
                   CupertinoIcons.clear_thick_circled,
                   size: 18.0,
-                  color: _kInactiveTextColor,
+                  color: placeholderStyle.color,
```

Comment:
I think we're on our own here since you can't change the placeholder's color on iOS itself. Since we don't have a precedent that says these colors are linked, I'd just leave this. (you can always use the suffix widget to put a custom button).